packages/core/src/workflows/workflow.ts

```diff
@@ -336,6 +380,379 @@ export function createStep<
     };
   }
 
+  // Handle Processor - wrap it as a workflow step
+  if (isProcessor(params)) {
+    const processor = params;
+    return {
+      // @ts-ignore - processor overload has specific id type
+      id: `processor:${processor.id}`,
+      description: processor.name ?? `Processor ${processor.id}`,
+      // @ts-ignore - Use discriminated union for input (better UI experience)
+      inputSchema: ProcessorStepSchema,
+      // @ts-ignore - Use flexible schema for output (allows any phase combination)
+      outputSchema: ProcessorStepOutputSchema,
+      execute: async ({
+        inputData,
+        requestContext,
+      }: {
+        inputData: z.infer<typeof ProcessorStepSchema>;
+        requestContext: RequestContext;
+      }): Promise<ProcessorStepOutput> => {
+        // Cast to output type for easier property access - the discriminated union
+        // ensures type safety at the schema level, but inside the execute function
+        // we need access to all possible properties
```

Comment:
I wonder if it'd be better if this whole execute function lived in the ../processors dir and just import here `executeProcessorWorkflowStep` or something. Just so the processor related code is separated from the rest of generic workflows stuff since a lot of this code used to live in the processor runner. Not a huge deal though.

Reply:
maybe even `ProcessorRunner.executeWorkflow({ inputData, requestContext })` ?